addons/controllers/awsebscsi/awsescsiconfig_utils.go

```diff
@@ -0,0 +1,61 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	clusterapiv1beta1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/vmware-tanzu/tanzu-framework/addons/pkg/constants"
+	csiv1alpha1 "github.com/vmware-tanzu/tanzu-framework/apis/addonconfigs/csi/v1alpha1"
+)
+
+// getOwnerCluster verifies that the AwsEbsCSIConfig has a cluster as its owner reference,
+// and returns the cluster. It tries to read the cluster name from the AwsEbsCSIConfig's owner reference objects.
+// If not there, we assume the owner cluster and AwsEbsCSIConfig always has the same name.
+func (r *AwsEbsCSIConfigReconciler) getOwnerCluster(ctx context.Context,
+	awsebsCSIConfig *csiv1alpha1.AwsEbsCSIConfig) (*clusterapiv1beta1.Cluster, error) {
+
+	logger := log.FromContext(ctx)
+	cluster := &clusterapiv1beta1.Cluster{}
+	// TODO donot support using addon name as cluster name
+	clusterName := awsebsCSIConfig.Name
+
+	// retrieve the owner cluster for the AwsEbsCSIConfig object
+	for _, ownerRef := range awsebsCSIConfig.GetOwnerReferences() {
+		if strings.EqualFold(ownerRef.Kind, constants.ClusterKind) {
+			clusterName = ownerRef.Name
+			break
+		}
+	}
+	if err := r.Client.Get(ctx, types.NamespacedName{Namespace: awsebsCSIConfig.Namespace, Name: clusterName}, cluster); err != nil {
+		if apierrors.IsNotFound(err) {
+			logger.Info(fmt.Sprintf("Cluster resource '%s/%s' not found", awsebsCSIConfig.Namespace, clusterName))
+			return nil, nil
+		}
+		logger.Error(err, fmt.Sprintf("Unable to fetch cluster '%s/%s'", awsebsCSIConfig.Namespace, clusterName))
+		return nil, err
+	}
+
+	return cluster, nil
+}
+
+// mapAwsEbsCSIConfigToDataValues maps AwsEbsCSIConfig CR to data values
+func (r *AwsEbsCSIConfigReconciler) mapAwsEbsCSIConfigToDataValues(ctx context.Context,
+	awsEbsCSIConfig *csiv1alpha1.AwsEbsCSIConfig,
+	cluster *clusterapiv1beta1.Cluster) (*DataValues, error) {
+
+	dvs := &DataValues{}
```

Comment:
Is there a template in TKR to provide default values?

Reply:
discussed with @wjun as well, this's a good point, hopefully, could use CRD mechanism to avoid incorrect values, but maybe will need a coming patch to follow up, given this is blocking other CSI packages, let me do more tests offline and file a new PR if needed, hope this makes sense.